docs/proposals/app-metrics.md

```diff
@@ -0,0 +1,172 @@
+
+### Universal
+At the mesh level, it will be possible to define the default exposed path and port at which metrics are going to be accessible.
+```
+type: Mesh
+name: default
+metrics:
+ enabledBackend: prometheus-1
+ backends:
+ - name: prometheus-1
+   type: prometheus
+   conf:
+    skipMTLS: false
+    port: 5670 # default exposed port with metrics 
+    path: /metrics
+    tags:   
+      kuma.io/service: dataplane-metrics
+    aggregate:
+     - name: app
+       port: 1236
+       path: /metrics
+       enabled: false
+     - name: opa
+       port: 12345
+       path: /stats
+       enabled: true
+```
+`Dataplane` configuration enables at which port/path `kuma-dp` is going to expose aggregated metrics from Envoy and other applications running within one workflow. Also, it's possible to override the `Mesh` configuration of endpoints to scrap by the `Dataplane` configuration. They are going to be identified by `name` and if `Mesh` and `Dataplane` configurations have definitions with the same name, then the `Dataplane` configuration has precedence before `Mesh` and is merged.
+```
+type: Dataplane
+mesh: default
+name: redis-1
+networking:
+ address: 192.168.0.1
+ inbound:
+ - port: 9000
+   servicePort: 1234
+   tags:
+     kuma.io/service: backend
+ metrics:
+   type: prometheus
+   conf:
+     skipMTLS: true
+     port: 1234
+     path: /non-standard-path
+     aggregate:
+     - name: app
+       enabled: false
+     - name: opa
+       port: 22222
+       path: /stats
+```
+
+Finally `kuma-dp` won't scrape metrics from `app`, because the user set it to false. Also, `opa` configuration is changed by the user so the configuration that takes an effect has changed. `Kuma-dp` is going to scrape only `opa` at port `22222`.
+ 
+### Kubernetes
+ 
+The same `Mesh` configuration works for K8s and Universal.
+```
+apiVersion: kuma.io/v1alpha1
+kind: Mesh
+metadata:
+ name: default
+spec:
+ metrics:
+   enabledBackend: prometheus-1
+   backends:
+   - name: prometheus-1
+     type: prometheus
+     conf:
+       skipMTLS: false
+       port: 5670
+       path: /metrics
+       tags: # tags that can be referred in Traffic Permission when metrics are secured by mTLS 
+         kuma.io/service: dataplane-metrics
+       aggregate:
+       - name: app
+         port: 1236
+         path: /metrics
+       - name: opa
+         port: 12345
+         path: /stats
+         # at the mesh it is always true
+         enabled: true 
+```
+In the case of Kubernetes configuration of endpoints to scrap is done by label `prometheus.metrics.kuma.io/aggregate-{name}-port` and `prometheus.metrics.kuma.io/aggregate-{name}-path`, where `{name}` is used for distinguish between config for different containers/apps running in the pod.
```

Comment:
Do you need to add `-enabled` too?

Reply:
`-enabled` is going to be by default. Only if you want to override value it is required to define